stbt-run

```diff
@@ -72,6 +80,35 @@ def _print_exc_utf8(file_=None):
         traceback._some_str = _old_some_str
 
 
+def _record_screenshot(exception):
+    import cv2
+
+    jpg = (args.write_thumbnail == 'always' or
+           (args.write_thumbnail == 'on-failure' and exception))
+    png = (args.write_screenshot == 'always' or
+           (args.write_screenshot == 'on-failure' and exception))
+
+    screenshot = getattr(exception, "screenshot", None)
+    if screenshot is None and stbt._dut._display:  # pylint: disable=protected-access
+        screenshot = stbt._dut._display.last_used_frame  # pylint: disable=protected-access
+    if screenshot is None:
+        screenshot = stbt._dut.get_frame()  # pylint: disable=protected-access
```

Comment:
This last clause (if `stbt._dut._display.last_used_frame` is None then grab a new frame by calling `get_frame()`) causes a regression in `test_invalid_source_pipeline`: Instead of the error message about the invalid source pipeline, you get a stack trace from `get_frame()`.